app/controllers/v2/repositories_controller.rb

```diff
@@ -0,0 +1,69 @@
+# frozen_string_literal: true
+
+module V2
+  # Handles all requests for repository CRUD operations
+  class RepositoriesController < ApplicationController
+    def index
+      @collection = Repository.all
+      render json: @collection, each_serializer: V2::RepositorySerializer
+    end
+
+    def show
+      retrieve_resource
+      render_resource
+    end
+
+    def create
```

Comment:
The only way I can think of is to use a handy superclass that defines all this for the typical case. I created a `ResourceController` similar to the basic idea of [inherited resources](https://github.com/josevalim/inherited_resources) but with much less functionality/magic. It gives us a simple way to stay DRY and a place where we can lookup defaults if we need to.